app/main.py

```diff
@@ -1,6 +1,31 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def wrapper(*args, **kwargs) -> Any:
+        flag = True
+        while flag:
+            data_iterable = tuple(
+                [item for item in args if isinstance(item, (int, str, float))]
+            )
+            if data_iterable not in cache_dict:
+                print("Calculating new result")
+                cache_dict[data_iterable] = func(*args, **kwargs)
+                return cache_dict[data_iterable]
+            else:
+                print("Getting from cache")
+                return cache_dict[data_iterable]
+
+    return wrapper
+
+
+@cache
+def long_time_func(a: int, b: int, c: int) -> int:  # NOQA VNE001
+    return (a ** b ** c) % (a * c)
+
+
+@cache
+def long_time_func_2(n_tuple: tuple, power: int) -> list:
```

Comment:
The `long_time_func_2` function is expected to return an `int`, but it currently returns a `list`. Please adjust the function to return an integer or update the return type annotation if a list is indeed the intended return type.